app/Http/Controllers/UpdatePasswordController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Interns2025b\Http\Controllers;
+
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Interns2025b\Http\Requests\UpdatePasswordRequest;
+use Symfony\Component\HttpFoundation\Response as Status;
+
+class UpdatePasswordController extends Controller
+{
+    public function updatePassword(UpdatePasswordRequest $request): JsonResponse
+    {
+        $user = Auth::user();
+        $validated = $request->validated();
+
+        if (Hash::check($validated["new_password"], $user->password)) {
```

Comment:
You can create `getNewPassword()` method in **UpdatePasswordRequest** instead of working with arrays